src/realm/object-store/object_accessor.hpp

```diff
@@ -281,33 +281,47 @@ Object Object::create(ContextType& ctx, std::shared_ptr<Realm> const& realm, Obj
         if (!primary_value && !is_nullable(primary_prop->type))
             throw MissingPropertyValueException(object_schema.name, primary_prop->name);
 
-        // When changing the primary key of a table, we remove the existing pk (if any), call
-        // the migration function, then add the new pk (if any). This means that we can't call
-        // create_object_with_primary_key(), and creating duplicate primary keys is allowed as
-        // long as they're unique by the end of the migration.
-        if (table->get_primary_key_column() == ColKey{}) {
-            REALM_ASSERT(realm->is_in_migration());
-            if (policy.update) {
-                if (auto key = get_for_primary_key_in_migration(ctx, *table, *primary_prop, *primary_value))
-                    obj = table->get_object(key);
+        if (object_schema.is_asymmetric) {
```

Comment:
I would not make a special case here, but have a test that `object_schema.is_asymmetric`excludes `policy.update`.